nixos/modules/services/security/crowdsec.nix

```
@@ -0,0 +1,945 @@
+          listen_addr = lib.mkDefault "127.0.0.1";
+          listen_port = lib.mkDefault 6060;
+        };
+        cscli = {
+          hub_branch = "v${cfg.package.version}";
+        };
+      };
+
+      environment = {
+        systemPackages = [ cscli ];
+      };
+
+      systemd.packages = [ cfg.package ];
+
+      systemd.timers.crowdsec-update-hub = lib.mkIf (cfg.autoUpdateService) {
+        description = "Update the crowdsec hub index";
+        wantedBy = [ "timers.target" ];
+        timerConfig = {
+          OnCalendar = "daily";
+          Persistent = "yes";
+          Unit = "crowdsec-update-hub.service";
+        };
+      };
+      systemd.services = {
+        crowdsec-update-hub = lib.mkIf (cfg.autoUpdateService) {
+          description = "Update the crowdsec hub index";
+          serviceConfig = {
+            Type = "oneshot";
+            User = cfg.user;
+            Group = cfg.group;
+            LimitNOFILE = 65536;
+            NoNewPrivileges = true;
+            LockPersonality = true;
+            RemoveIPC = true;
+            ReadWritePaths = [
+              rootDir
+              confDir
+            ];
+            ProtectSystem = "strict";
+            PrivateUsers = true;
+            ProtectHome = true;
+            PrivateTmp = true;
+            PrivateDevices = true;
+            ProtectHostname = true;
+            UMask = "0077";
+            ProtectKernelTunables = true;
+            ProtectKernelModules = true;
+            ProtectControlGroups = true;
+            ProtectProc = "invisible";
+            SystemCallFilter = [
+              " " # This is needed to clear the SystemCallFilter existing definitions
+              "~@reboot"
+              "~@swap"
+              "~@obsolete"
+              "~@mount"
+              "~@module"
+              "~@debug"
+              "~@cpu-emulation"
+              "~@clock"
+              "~@raw-io"
+              "~@privileged"
+              "~@resources"
+            ];
+            CapabilityBoundingSet = [
+              " " # Reset all capabilities to an empty set
+            ];
+            RestrictAddressFamilies = [
+              " " # This is needed to clear the RestrictAddressFamilies existing definitions
+              "none" # Remove all addresses families
+              "AF_UNIX"
+              "AF_INET"
+              "AF_INET6"
+            ];
+            DevicePolicy = "closed";
+            ProtectKernelLogs = true;
+            SystemCallArchitectures = "native";
+            RestrictNamespaces = true;
+            RestrictRealtime = true;
+            RestrictSUIDSGID = true;
+            ExecStart = "${lib.getExe cscli} --error hub update";
+            ExecStartPost = "systemctl reload crowdsec.service";
+          };
+        };
+
+        crowdsec = {
+          description = "CrowdSec is a free, modern & collaborative behavior detection engine, coupled with a global IP reputation network.";
```

```suggestion
          description = "CrowdSec is a free, modern & collaborative behavior detection engine, coupled with a global IP reputation network.";
          description = "CrowdSec agent";
```

This is intended to be the name of the service.

As-is, I’m getting 
```
systemd[1]: Failed to start CrowdSec is a free, modern & collaborative behavior detection engine, coupled with a global IP reputation network..
```
in my logs while setting this up to test.

The [upstream service definition][1] uses ‘Crowdsec agent', so let’s do the same.

[1]: https://github.com/crowdsecurity/crowdsec/blob/76ea87c6bae9cdcbaec6fdb5f56962f64affb701/config/crowdsec.service